docs/config.md

```diff
@@ -663,15 +663,27 @@ java -Dspring.config.location=file:/path/to/your/myconfig.yaml -jar promregator-
 
+#### Option "promregator.authenticator.oauth2xsuaa.xsuaaServiceURL"
+Like `promregator.authenticator.oauth2xsuaa.tokenServiceURL`, but without trailing `/oauth/token`.
+
+#### Option "promregator.authenticator.oauth2xsuaa.xsuaaServiceCertURL"
+Specifies the URL of the OAuth2 endpoint for certificate based authentication, which contains the token service of your authorization server in case of global authentication. Needs to be provided without a path.
+
 #### Option "promregator.authenticator.oauth2xsuaa.client_id" (mandatory, if using promregator.authenticator.type=OAuth2XSUAA)
 Specifies the client identifier (a.k.a. "client_id") which shall be used during the OAuth2 request based on the Grant Type Client Credentials flow in case of global authentication.
 
-#### Option "promregator.authenticator.oauth2xsuaa.client_secret" (mandatory, if using promregator.authenticator.type=OAuth2XSUAA)
-Specifies the client secret (a.k.a. "client_secret") which shall be used during the OAuth2 request based on the Grant Type Client Credentials flow in case of global authentication.
+#### Option "client_certificates" (mandatory for certificate-based authentication, if using promregator.authenticator.type=OAuth2XSUAA)
+Specifies the certificate chain which shall be used during the OAuth2 request. Omitted when client secret-based authentification is used.
+
+#### Option "promregator.authenticator.oauth2xsuaa.client_secret" (mandatory for client_secret-based authentication, if using promregator.authenticator.type=OAuth2XSUAA)
+Specifies the client secret (a.k.a. "client_secret") which shall be used during the OAuth2 request based on the Grant Type Client Credentials flow in case of global authentication. Omitted, if certificate based authentication is used.
+
+#### Option "client_key" (mandatory for certificate based authentication, if using promregator.authenticator.type=OAuth2XSUAA)
+Specifies the private key which shall be used during the OAuth2 request. Omitted when client secret-based authentification is used.
 
 *WARNING!* 
-Due to security reasons, it is *neither* recommended to store this value in your YAML file, nor to put it into the command line when starting Promregator.
-Instead it is suggested to set the identically named environment variable `promregator.authenticator.oauth2xsuaa.client_secret` when starting the application.
+Due to security reasons, it is *neither* recommended to store the value for `client_secret` and `client_key` in your YAML file, nor to put it into the command line when starting Promregator.
+Instead it is suggested to set the identically named environment variables `PROMREGATOR_AUTHENTICATOR_OAUTH2XSUAA_CLIENT_SECRET` and `PROMREGATOR_AUTHENTICATOR_OAUTH2XSUAA_CLIENT_KEY` when starting the application.
```

Comment:
I think we have to split and duplicate this sentence here and also add it to the `client_secret` section.
There will be users out there which will only read one variant only.

Reply:
done